elastica/rod/factory_function.py

```diff
@@ -216,7 +215,43 @@ def allocate(
         )
 
     # Shear/Stretch matrix
-    shear_modulus = youngs_modulus / (poisson_ratio + 1.0)
+    if kwargs.__contains__("shear_modulus"):
+        shear_modulus = kwargs.get("shear_modulus")
+        if kwargs.__contains__("poisson_ratio"):
+            poisson_ratio = kwargs.get("poisson_ratio")
+            message = (
+                " Poisson ratio ( "
+                + str(poisson_ratio)
+                + " ) given in kwargs is not used. \n"
+                + "Since shear modulus ( "
+                + str(shear_modulus)
+                + " ) is given in kwargs. \n"
+            )
+            warnings.warn(message, category=UserWarning)
+    else:
+        if kwargs.__contains__("poisson_ratio"):
+            poisson_ratio = kwargs.get("poisson_ratio")
+        else:
+            message = "Shear modulus or poisson ratio cannot be found in kwargs. Poisson ratio is taken as 0.5"
+            warnings.warn(message, category=UserWarning)
+            poisson_ratio = 0.5
+
+        shear_modulus = youngs_modulus / (poisson_ratio + 1.0)
+
+        message = (
+            "Shear modulus cannot be found in kwargs. \n"
+            "Poisson ratio "
```

Comment:
I would suggest the warning being:
 "The given Poisson ratio of "+ str(poisson_ratio) + " is used only to compute a shear modulus of " + str(shear_modulus) + ", " "using the equation: shear_modulus = youngs_modulus / (poisson_ratio + 1.0). Use of a Poisson ratio will be depreciated in a future release. It is encouraged that you discontinue using a Poisson ratio and instead directly provide the shear_modulus"